plugin/src/test/java/io/jenkins/plugins/casc/misc/JenkinsConfiguredWithReadmeRule.java

```diff
@@ -0,0 +1,103 @@
+        if (Objects.nonNull(configuredWithReadme)) {
+
+            final Class<?> clazz = env.description().getTestClass();
+            final String[] resource = configuredWithReadme.value();
+
+            // TODO: transform from `value` to Code blocks
+
+            final List<String> configs = Arrays.stream(resource)
+                .map(s -> clazz.getClassLoader().getResource(s).toExternalForm())
+                .collect(Collectors.toList());
+
+            try {
+                ConfigurationAsCode.get().configure(configs);
+            } catch (Throwable t) {
+                if (!configuredWithReadme.expected().isInstance(t)) {
+                    throw new AssertionError("Unexpected exception ", t);
+                } else {
+                    if (!StringUtils.isBlank(configuredWithReadme.message())) {
+                        boolean match = new StringContains(configuredWithReadme.message())
+                            .matches(t.getMessage());
+                        if (!match) {
+                            throw new AssertionError(
+                                "Exception did not contain the expected string: "
+                                    + configuredWithReadme.message() + "\nMessage was:\n" + t
+                                    .getMessage());
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private ConfiguredWithReadme getConfiguredWithReadme() {
+        ConfiguredWithReadme configuredWithReadme = env.description()
+            .getAnnotation(ConfiguredWithReadme.class);
+        if (Objects.nonNull(configuredWithReadme)) {
+            return configuredWithReadme;
+        }
+        for (Field field : env.description().getTestClass().getFields()) {
+            if (field.isAnnotationPresent(ConfiguredWithReadme.class)) {
+                int m = field.getModifiers();
+                Class<?> clazz = field.getType();
+                if (isPublic(m) && isStatic(m) &&
+                    clazz.isAssignableFrom(JenkinsConfiguredWithReadmeRule.class)) {
+                    configuredWithReadme = field.getAnnotation(ConfiguredWithReadme.class);
+                    if (Objects.nonNull(configuredWithReadme)) {
+                        return configuredWithReadme;
+                    }
+                } else {
+                    throw new IllegalStateException(
+                        "Field must be public static JenkinsConfiguredWithReadmeRule");
+                }
+            }
+        }
+        return null;
+    }
+
+    //TODO: Looks like API defect, exception should be thrown
+    /**
+     * Exports the Jenkins configuration to a string.
+     * @return YAML as string
+     * @param strict Fail if any export operation returns error
+     * @throws Exception Export error
+     * @throws AssertionError Failed to export the configuration
+     * @since 1.25
+     */
+    public String exportToString(boolean strict) throws Exception {
```

Comment:
should be strict by default in new code I assume?

Reply:
I just refactored it: 
- https://github.com/jenkinsci/configuration-as-code-plugin/pull/1055/commits/cdde7b89c04b7aec10e62fd14f1f65f32d5f74ee